### ISSUE TITLE
Bad UI styling with some GTK themes

### ISSUE THREAD

**Issue by timvisee** [mozilla-lockbox/lockbox-extension#264]
# Problem description
The styling of some UI components, such as buttons or links seems to be quite bad in combination with some GTK themes on a user's system. Some UI components become nearly invisible making them hard to see. The styling of some other components is simply, ugly.

In this case, I'm running a dark system-wide GTK theme. Firefox seems to use some of the default colors of GTK themes used on a Linux system. This causes problems with the current UI theme of LockBox. Therefore I'm also sure similar issues might happen when a user has a different GTK theme enabled.

# Screenshots
Let's take a look at some examples below, or view the [full-page screenshot](https://uploads.timvisee.com/public/firefox-lockbox-full-29e87ab8.png).

### Sidebar
![Sidebar screenshot](https://uploads.timvisee.com/public/firefox-lockbox-sidebar-e68eeab5.png)
- The search bar (and _Clear_ button) are unreadable.
- The entry selection color is weird.

### Edit
![Edit screenshot](https://uploads.timvisee.com/public/firefox-lockbox-edit-d4dcdbaf.png)
- Fields are unreadable.
- The _Show_ and _Cancel_ button are unreadable.

### Top bar
![Top bar screenshot](https://uploads.timvisee.com/public/firefox-lockbox-topbar-1cf5b557.png)
- Action buttons in the top bar are unreadable.

### Dialog box
![Dialog box screenshot](https://uploads.timvisee.com/public/firefox-lockbox-dialog-49bb7c9a.png)
- Dialog box backgrounds are dark.
- The _Cancel_ button is unreadable.

# Things I've tried
- Changing the Firefox theme in the customization menu does not have any effect.
- Disabling GTK themes when starting Firefox solves the issue.

# Possible solution
It looks like a custom color is set for input field backgrounds, but not for their text. The text colors are inherited from the user's GTK theme, if available. This seems to be causing issues when using a dark GTK theme.

Explicitly setting the text color on input fields (to black for example) would probably fix this issue no matter what theme is used.

I'm sorry if changes like this fall out of the scope of this project, and should be applied to some UI library instead. In that case I'll be sure to post this issue in that project instead.

# System
- OS: Ubuntu 17.04
- Firefox: Nightly 58.0a1 (2017-10-29) (64-bit)
- Firefox theme: Dark (built-in)
- Firefox plugins: _No theme chaining plugins, thus irrelevant_
- GTK theme: Arc-Dark

**Comment by devinreams** [comment]
Thanks for the report @timvisee. We're spending some time on this next sprint to revisit all the styles we're inheriting (browser styles for extensions) and make improvements that more closely match [Photon](http://design.firefox.com/photon/welcome.html).

While this was in line with what we were already planning, we certainly appreciate the detailed issue and contribution. 🙇

**Comment by timvisee** [comment]
@devinreams That's brilliant! I'm sure a revisit on the styles might fix it.

I've already been working on a simple fix though, that I've just submitted as pull request before I saw your message. Maybe it's useful for now. I'll leave it open for you to decide whether it's useful 😃 .

**Comment by jimporter** [comment]
https://github.com/mozilla-lockbox/lockbox-extension/pull/295 *should* fix all this.

**Comment by jimporter** [comment]
@timvisee I just landed #295, so hopefully everything looks good on your end. We should be releasing a new version of the add-on with this fix in a couple of days.

**Comment by timvisee** [comment]
Hello! Sorry for not being a available these few days.

I've just cloned the newest commit 3d5c28488fb96f348049abb16f13c6718d28f7ea@master, and have compiled it. Here are my results:

---

The improvements are great, although there do seem to be some things that aren't fully (visually) working on my system yet. Most of the problems with textboxes having hard-to-read text colors have been fixed. The password fields still have this issue though (both the placeholder, the balls inside them, and the text when the _show_ button is pressed). Next to the password box, the search field does seem to have the same issue.

Here are two screenshots showing these issues:
![New Firefox Lockbox welcome screen](https://uploads.timvisee.com/public/firefox-lockbox-welcome-6c02a9fc.png)

![New Firefox Lockbox edit screen](https://uploads.timvisee.com/public/firefox-lockbox-edit-abb560c3.png)

I've also noticed that `textarea` fields inside the entry edit screen have a slightly lighter blueish border color when focused. This is not a problem at all of course, but I thought it would be nice to mention it at the least.

---

Things like selection colors, buttons, header buttons, dialogs and whatnot have all been fixed. 😄 

To summarize what still needs work:
- Password fields text color
- Search box text color

Thank you for the improvements so far, truly awesome!

**Comment by devinreams** [comment]
Thanks for checking @timvisee. Would you mind if we...

1. spin these two items (password input color, search input color) into the follow-on Photon work at #331?
2. close your open PR #268 as "superseded" by this work?

**Comment by timvisee** [comment]
@devinreams Yes, that is perfectly fine! Thank you for looking into it.

**Comment by jimporter** [comment]
@devinreams Don't split it into two items; it's literally the same block of CSS shared by both components.

**Comment by devinreams** [comment]
>Yes, that is perfectly fine! Thank you for looking into it.

Thanks again @timvisee. I tacked another "task" onto our list there and will address it all at once in the next sprint. I'll close this as duplicate just to tidy up. We appreciate your quick and thorough feedback.

Duplicate of #331

**Comment by devinreams** [comment]
Whoop sorry, I now see @jimporter tackling this in a commit. 😅
